### PR TITLE
cargo-fmt: don't return zero on failure

### PR DESCRIPTION
**Pull request by matklad**
Previously `cargo-fmt` always returned `0` exit status. Now it returns `1` if command line flags are wrong and forwards status code of `rustfmt` oherwise

### CODE REVIEW

File: src/bin/cargo-fmt.rs [hunk 1]
```diff
@@ -42,13 +52,26 @@ fn main() {
...
+
+    match format_crate(&opts, verbose, quiet) {
+        Err(e) => {
+            print_usage(&opts, &e.to_string());
+            return failure;
+        }
+        Ok(status) => {
+            if !status.success() {
+                return status.code().unwrap_or(failure);
+            }
```
Comment (nrc) [inline src/bin/cargo-fmt.rs]: style nit: avoid the return here by using if ... else (and above, prefer `failure` to `return failure;`

**Comment by nrc** [comment]
Thanks for the PR! Looks good, r+ with the style nit addressed.

File: src/bin/cargo-fmt.rs (hunk 1 above)
Comment (matklad) [inline src/bin/cargo-fmt.rs]: Yeah, mixing `match` and early return on error was a terrible choice :) Fixed and rebased.

**Comment by matklad** [comment]
wait a moment, there is an unused variable left!

**Comment by matklad** [comment]
>wait a moment, there is an unused variable left!

Fixed and slapped `#[deny(warnings)]` to avoid future embarrassments.

**Comment by nrc** [comment]
Thanks for the changes!

**Pull request merged**